uni/lib/view/schedule/widgets/schedule_slot.dart

```diff
@@ -104,21 +108,38 @@ class ScheduleSlot extends StatelessWidget {
 
 class SubjectButtonWidget extends StatelessWidget {
   const SubjectButtonWidget({required this.occurrId, super.key});
+
   final int occurrId;
+  CourseUnit correspondingCourseUnit(BuildContext context) {
+    final courseUnits =
+        Provider.of<ProfileProvider>(context, listen: false).state!.courseUnits;
+    final notFound =
+        CourseUnit(abbreviation: 'NF', name: 'not found', occurrId: 0);
+    final correspondingCourseUnit = courseUnits.firstWhere(
+      (courseUnit) => courseUnit.occurrId == occurrId,
+      orElse: () => notFound,
+    );
 
-  String toUcLink(int occurrId) {
-    const faculty = 'feup'; // should not be hardcoded
-    return '${NetworkRouter.getBaseUrl(faculty)}'
-        'UCURR_GERAL.FICHA_UC_VIEW?pv_ocorrencia_id=$occurrId';
+    return correspondingCourseUnit;
   }
 
-  Future<void> _launchURL(BuildContext context) async {
-    final url = toUcLink(occurrId);
-    await launchUrlWithToast(context, url);
+  void _launchURL(BuildContext context) {
```

Comment:
Please change this name to be accurate. Something like `_launchUcPage` should be good.